store/store.go

```diff
@@ -108,11 +117,13 @@ func newStore[H header.Header[H]](ds datastore.Batching, opts ...Option) (*Store
 		writesDn:    make(chan struct{}),
 		pending:     newBatch[H](params.WriteBatchSize),
 		Params:      params,
+
+		knownHeaders: make(map[uint64]H),
```

Comment:
I think this map can be simplified to `map[uint64][]byte`. The whole point of knowHeaders is to correctly update the head. But at the same time all headers are verified and added to the batch, so we can get the header from there by calling `GetByHeight/GetByHash`.

Reply:
Good point but there is a problem. We're trying to advance `writeHead` before batch is *really* written on disk. Header may not exists yet in batch/pending so `GetByHeight` will return error.

@renaynay AFAIR we agreed to advance `writeHead` when we submit headers to flush worker, based on paragraph above should we change the moment when we want to update `writeHead` ? If not than I think this comment can be resolved as not feasible.

Reply:
I see, thanks. 

> should we change the moment when we want to update writeHead ?

I don't think it's neccessary.

Reply:
We only want to update writeHead when the headers have actually been written to disk and the contiguous head has advanced

Reply:
Fixed